[CI] DocsClientYamlTestSuiteIT test {yaml=reference/indices/get-data-stream/line_271} failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/wcv26mumyo3be/tests/:docs:yamlRestTest/org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT/test%20%7Byaml=reference%2Findices%2Fget-data-stream%2Fline_271%7D
**Reproduction line:**
```
./gradlew ':docs:yamlRestTest' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/indices/get-data-stream/line_271}" -Dtests.seed=47E87BD7A1BBF766 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=zh-TW -Dtests.timezone=Africa/Bissau -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT&tests.test=test%20%7Byaml%3Dreference/indices/get-data-stream/line_271%7D


**Failure excerpt:**
```
java.lang.AssertionError: Failure at [reference/indices/get-data-stream:102]: Expected a map containing
          data_streams: a list containing
                       0: a map containing
                generation: <2>
                  template: "my-index-template"
next_generation_managed_by: "Index Lifecycle Management"
                prefer_ilm: <true>
                    hidden: <false>
                     _meta: a map containing
               my-meta-field: "foo"
           timestamp_field: a map containing
                        name: "@timestamp"
                   indices: a list containing
                           0: a map containing
                    index_uuid: "MdcayGgdSLW_Y1RkGPU0tA"
                    prefer_ilm: <true>
                    managed_by: "Index Lifecycle Management"
                    ilm_policy: "my-lifecycle-policy"
                    index_name: ".ds-my-data-stream-2023.11.16-000001"
                           1: a map containing
                    index_uuid: "cNRARRdiSQWZVLPKPlFhGw"
                    prefer_ilm: <true>
                    managed_by: "Index Lifecycle Management"
                    ilm_policy: "my-lifecycle-policy"
                    index_name: ".ds-my-data-stream-2023.11.16-000002"
                    system: <false>
      allow_custom_routing: <false>
           failure_indices: expected a list but was <missing>
                      name: "my-data-stream"
             failure_store: expected <false> but was <missing>
                ilm_policy: "my-lifecycle-policy"
                replicated: <false>
                    status: "YELLOW"
                       1: a map containing
                generation: <1>
                  template: "my-index-template"
next_generation_managed_by: "Index Lifecycle Management"
                prefer_ilm: <true>
                    hidden: <false>
                     _meta: a map containing
               my-meta-field: "foo"
           timestamp_field: a map containing
                        name: "@timestamp"
                   indices: a list containing
                           0: a map containing
                    index_uuid: "p14w1in0QCSzqneb4l6TRA"
                    prefer_ilm: <true>
                    managed_by: "Index Lifecycle Management"
                    ilm_policy: "my-lifecycle-policy"
                    index_name: ".ds-my-data-stream-two-2023.11.16-000001"
                    system: <false>
      allow_custom_routing: <false>
           failure_indices: expected a list but was <missing>
                      name: "my-data-stream-two"
             failure_store: expected <false> but was <missing>
                ilm_policy: "my-lifecycle-policy"
                replicated: <false>
                    status: "YELLOW"

  at __randomizedtesting.SeedInfo.seed([47E87BD7A1BBF766:CFBC440D0F479A9E]:0)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:572)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:524)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

  Caused by: java.lang.AssertionError: Expected a map containing
            data_streams: a list containing
                         0: a map containing
                  generation: <2>
                    template: "my-index-template"
  next_generation_managed_by: "Index Lifecycle Management"
                  prefer_ilm: <true>
                      hidden: <false>
                       _meta: a map containing
                 my-meta-field: "foo"
             timestamp_field: a map containing
                          name: "@timestamp"
                     indices: a list containing
                             0: a map containing
                      index_uuid: "MdcayGgdSLW_Y1RkGPU0tA"
                      prefer_ilm: <true>
                      managed_by: "Index Lifecycle Management"
                      ilm_policy: "my-lifecycle-policy"
                      index_name: ".ds-my-data-stream-2023.11.16-000001"
                             1: a map containing
                      index_uuid: "cNRARRdiSQWZVLPKPlFhGw"
                      prefer_ilm: <true>
                      managed_by: "Index Lifecycle Management"
                      ilm_policy: "my-lifecycle-policy"
                      index_name: ".ds-my-data-stream-2023.11.16-000002"
                      system: <false>
        allow_custom_routing: <false>
             failure_indices: expected a list but was <missing>
                        name: "my-data-stream"
               failure_store: expected <false> but was <missing>
                  ilm_policy: "my-lifecycle-policy"
                  replicated: <false>
                      status: "YELLOW"
                         1: a map containing
                  generation: <1>
                    template: "my-index-template"
  next_generation_managed_by: "Index Lifecycle Management"
                  prefer_ilm: <true>
                      hidden: <false>
                       _meta: a map containing
                 my-meta-field: "foo"
             timestamp_field: a map containing
                          name: "@timestamp"
                     indices: a list containing
                             0: a map containing
                      index_uuid: "p14w1in0QCSzqneb4l6TRA"
                      prefer_ilm: <true>
                      managed_by: "Index Lifecycle Management"
                      ilm_policy: "my-lifecycle-policy"
                      index_name: ".ds-my-data-stream-two-2023.11.16-000001"
                      system: <false>
        allow_custom_routing: <false>
             failure_indices: expected a list but was <missing>
                        name: "my-data-stream-two"
               failure_store: expected <false> but was <missing>
                  ilm_policy: "my-lifecycle-policy"
                  replicated: <false>
                      status: "YELLOW"

    at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:82)
    at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:65)
    at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:94)
    at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:65)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:552)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:524)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke(Method.java:580)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:1583)

```

--- a/docs/build.gradle
+++ b/docs/build.gradle
@@ -72,6 +72,10 @@ testClusters.matching { it.name == "yamlRestTest"}.configureEach {
     setting 'xpack.license.self_generated.type', 'trial'
     setting 'indices.lifecycle.history_index_enabled', 'false'
     keystorePassword 'keystore-password'
+    if (BuildParams.isSnapshotBuild() == false) {
+//      systemProperty 'es.failure_store_feature_flag_enabled', 'true'
+      requiresFeature 'es.failure_store_feature_flag_enabled', new Version(8, 12, 0)
+    }
   }
 
   // debug ccr test failures:

--- a/docs/build.gradle
+++ b/docs/build.gradle
@@ -73,7 +73,6 @@ testClusters.matching { it.name == "yamlRestTest"}.configureEach {
     setting 'indices.lifecycle.history_index_enabled', 'false'
     keystorePassword 'keystore-password'
     if (BuildParams.isSnapshotBuild() == false) {
-//      systemProperty 'es.failure_store_feature_flag_enabled', 'true'
       requiresFeature 'es.failure_store_feature_flag_enabled', new Version(8, 12, 0)
     }
   }

